test/support/generator/create-facility-covenant-generator.ts

```diff
@@ -0,0 +1,117 @@
+import { ENUMS, PROPERTIES } from '@ukef/constants';
+import { UkefId } from '@ukef/helpers';
+import { AcbsCreateFacilityCovenantRequestDto } from '@ukef/modules/acbs/dto/acbs-create-facility-covenant-request.dto';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { CreateFacilityCovenantRequestDto, CreateFacilityCovenantRequestItem } from '@ukef/modules/facility-covenant/dto/create-facility-covenant-request.dto';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class CreateFacilityCovenantGenerator extends AbstractGenerator<CreateFacilityCovenantRequestItem, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): CreateFacilityCovenantRequestItem {
+    const possibleCovenantTypeCodes = Object.values(ENUMS.COVENANT_TYPE_CODES);
+
+    return {
+      facilityIdentifier: this.valueGenerator.ukefId(),
+      covenantIdentifier: this.valueGenerator.ukefId(), // TODO APIM-106: is this right to use ukefId()?
```

Comment:
For facility id we started to use function facilityId , but it is just alias for ukefId and returns ukefId content type.

Covenant ids are using slightly different pattern. You can look at content type UkefCovenantId, we don't have generator for it, but anything that starts with 0000 and has 10 digits should be fine.

Reply:
I've just had a look at the Postman collection again and the examples there don't use 10 digits, e.g., the GET covenants response is as follows:

![image](https://user-images.githubusercontent.com/116292912/233980298-4be6c65c-6e8c-4f4b-adbc-baa6e89f0733.png)

According to the ACBS docs, it can be any string length 0 to 10 inclusive. Do we want to restrict to 10 digits beginning with 0000, or are we happy to allow anything that ACBS accepts?

Reply:
We had discussion with Abhi last week about this. He is getting ids from number generator, so he will continue to use "10 digits beginning with 0000". 

Let's add this validation to be consistent, but it could be in separate PR.

Reply:
I've added this validation, a reusable test for it, and a `ukefCovenantId` generator.